Mathlib/Combinatorics/SimpleGraph/Triangle/Counting.lean

```diff
@@ -0,0 +1,178 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Combinatorics.SimpleGraph.Clique
+import Mathlib.Combinatorics.SimpleGraph.Regularity.Uniform
+import Mathlib.Data.Real.Basic
+import Mathlib.Tactic.Linarith
+
+/-!
+# Triangle counting lemma
+
+In this file, we prove the triangle counting lemma.
+
+## References
+
+[Yaël Dillies, Bhavik Mehta, *Formalising Szemerédi’s Regularity Lemma in Lean*][srl_itp]
+-/
+
+-- TODO: This instance is bad because it creates data out of a Prop
+attribute [-instance] decidableEq_of_subsingleton
+
+open Finset Fintype
+open scoped BigOperators
+
+variable {α : Type*} (G G' : SimpleGraph α) [DecidableRel G.Adj] {ε : ℝ} {s t u : Finset α}
+
+namespace SimpleGraph
+
+/-- The pairs of vertices whose density is big. -/
+private noncomputable def badVertices (ε : ℝ) (s t : Finset α) :=
```

Comment:
In this name (and throughout this file) you seem to be using 'vertices' to refer to parts of the partition of the big graph G. I understand why conceptually, but given how you've defined reduced graphs maybe would be less confusing to refer to these as parts instead? e.g. call this definition badParts and so on.

Reply:
I actually think I mean honest to god vertices here? I've updated the docstring to make things clearer.